src/completion.ts

```diff
@@ -575,14 +648,14 @@ class MdCompletionItemProvider implements CompletionItemProvider {
 
                 res(headingCompletions);
             });
-        } else if (/\[[^\]]*?\]\([^\)]*$/.test(lineTextBefore)) {
+        } else if (/\[[^\]]*?\]((\([^\]*])|\:(\s?\S*))$/.test(lineTextBefore)) {
             /* ┌────────────┐
                │ File paths │
                └────────────┘ */
             //// Should be after anchor completions
             if (workspace.getWorkspaceFolder(document.uri) === undefined) return [];
 
-            const typedDir = lineTextBefore.substr(lineTextBefore.lastIndexOf('](') + 2);
+            const typedDir = lineTextBefore.match(/(?<=((\]\()|(\]\:\s?)))\S*$/)[0];
```

Comment:
This PR is not OK.

I observed strange regressions.

To troubleshoot, I rewrote the "Reference link labels" part and it works now.

But I'm still trying to understand what's wrong with this piece.